core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalFile.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.ufs;
+
+import alluxio.util.URIUtils;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.UUID;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A simple data structure that represents a journal file.
+ * There are four types of files (SN is short for sequence number):
+ * 1. Temporary file: the file name is a random UUID.
+ * 2. Completed log file: [start SN]-[end SN] (the start SN is inclusive, the end SN is exclusive).
+ * 3. Incomplete log file (a.k.a the current log): [start SN]-[LONG.MAX_VALUE] (the start SN is
+ *    inclusive).
+ * 4. Checkpoint file: [0]-[end SN] (the end SN is exclusive).
+ *
+ * This data structure implements {@link Comparable} such that journal files can be sorted by the
+ * end SNs.
+ */
+@ThreadSafe
+final class UfsJournalFile implements Comparable<UfsJournalFile> {
```

Comment:
The naming is confusing to me. It is really metadata about files, right? Also, the `create*` methods look like files are created, but really, just the data structure is being created, right?